readthedocs/templates/sphinx/layout.html

```diff
@@ -3,21 +3,27 @@
 {% block extrahead %}
 {{ super() }}
 <!-- RTD Extra Head -->
-{% set ending = "/" if builder == "readthedocsdirhtml" else ".html" %}
-{% set canonical_page = pagename + ending %}
+{% if pagename == "index" %}
+    {% set canonical_page = "" %}
+{% else %}{% if pagename.endswith("/index") %}
```

Comment:
True, I'm coming from the past :) There we didn't had elif's. At least in Django, but this is Jinja2 anyways... Is fixed in the latest version of this branch.